### ISSUE TITLE
DarkPlaces geometry in wrong place

### ISSUE THREAD

**Issue by icecream95** [ptitSeb/box86#158]
Windows [DarkPlaces](https://icculus.org/twilight/darkplaces/files/darkplacesenginewindowsonly20140513.zip) runs with wine, but everything is in the wrong place:

This also affects movement, not just rendering. Turning with the cursor keys can only be done through about 90 degrees, before the direction of turning reverses.

![dp](https://user-images.githubusercontent.com/29191311/88793704-5a44c000-d1f1-11ea-909a-e6b9380ca622.png)

**Comment by ptitSeb** [comment]
Yes, indeed, strange geometry.

I'll try first to check if it"s a bug in a Dynarec Opcode.

**Comment by ptitSeb** [comment]
I fixed interpretor that were freezing with wine, so I could then run DarkPlaces without Dynarec... But same geometry issue, so the error is not Dynarec only.

**Comment by icecream95** [comment]
Almost a year later, and I've finally managed to find the cause of this issue.

I opened up DarkPlaces's `exe` in Cutter (analysis takes too long with Ghidra...), and looked through some functions using `sin` and `cos` to see how those functions were actually implemented (I suspected that either `asin`/`acos` were being no-ops somehow, or `sin`/`cos` was being executed too many times).

![asm](https://user-images.githubusercontent.com/29191311/126461847-2bf950cd-6e88-4e97-8b72-ed0c270a402f.png)

Getting to the actual function with the x87 trig instructions for `cos`, I noticed that `fcos` appeared twice. This case was predicated on a floating point status flag (`C2` it seems), so I tried making `fnstsw` return 0, and...

![dp](https://user-images.githubusercontent.com/29191311/126463397-c688efef-5bdf-4577-b20f-ff173b166766.png)

Everything rendered perfectly!

Doing more research on floating point status flags I found that `fsin`, `fcos` and `fsincos` should clear condition code bit `C2` when successful, but box86 was not doing that.

After a bit of patching, box86 was clearing the condition codes properly so I could remove the `fnstsw` hack and everything still worked fine:

[fpcond.patch](https://github.com/ptitSeb/box86/files/6854381/fpcond.patch.txt)
